terraform/aws/implementation/modules/eks/main.tf

```diff
@@ -1,32 +1,298 @@
+  kubernetes {
+    host                   = module.eks-cluster.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks-cluster.cluster_certificate_authority_data)
+    token                  = data.aws_eks_cluster_auth.eks.token
   }
+}
 
-  # Ensure that IAM Role permissions are created before and deleted after EKS Cluster handling.
-  # Otherwise, EKS will not be able to properly delete EKS managed EC2 infrastructure such as Security Groups.
-  depends_on = [
-    aws_iam_role_policy_attachment.amazonEKSClusterPolicy,
-    aws_iam_role_policy_attachment.amazonEKSVPCResourceController,
-  ]
+provider "kubernetes" {
+  host                   = module.eks-cluster.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks-cluster.cluster_certificate_authority_data)
+  token                  = data.aws_eks_cluster_auth.eks.token
+}
+
+resource "aws_iam_policy" "load_balancer_controller" {
+  name        = "AWSLoadBalancerControllerIAMPolicy"
+  description = "Policy for ALB Ingress Controller"
+  policy      = data.aws_iam_policy_document.load_balancer_controller.json
+}
+
+resource "kubectl_manifest" "cluster_role" {
+  yaml_body = data.kubectl_path_documents.cluster_role.documents[0]
+}
+
+resource "kubectl_manifest" "cluster_role_binding" {
+  yaml_body = data.kubectl_path_documents.cluster_role_binding.documents[0]
+}
+
+locals {
+  account_id           = data.aws_caller_identity.current.account_id
+  oidc_provider        = module.eks-cluster.oidc_provider
+  namespace            = "kube-system"
+  service_account_name = "aws-load-balancer-controller"
+}
+
+resource "aws_iam_role" "eks_service_account" {
+  name = "AmazonEKSLoadBalancerControllerRole"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Effect = "Allow"
+        Principal = {
+          Federated = "arn:aws:iam::${local.account_id}:oidc-provider/${local.oidc_provider}"
+        }
+        Condition = {
+          StringEquals = {
+            "${local.oidc_provider}:aud" = "sts.amazonaws.com"
+            "${local.oidc_provider}:sub" = "system:serviceaccount:${local.namespace}:${local.service_account_name}"
+          }
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "load_balancer_controller" {
+  role       = aws_iam_role.eks_service_account.name
+  policy_arn = aws_iam_policy.load_balancer_controller.arn
+}
+
+resource "kubectl_manifest" "service_account" {
+  yaml_body = data.kubectl_file_documents.service_account.documents[0]
+}
+
+# kubectl auth to EKS
+resource "terraform_data" "kubeconfig" {
+  depends_on = [module.eks-cluster, kubectl_manifest.cluster_role_binding, kubectl_manifest.service_account]
+
+  provisioner "local-exec" {
+    command = "aws eks update-kubeconfig --name ${module.eks-cluster.cluster_name} --region ${var.region}"
+  }
+}
+
+# cert-manager
+
+# resource "helm_release" "cert_manager" {
```

Comment:
Can this commented section be deleted?

Reply:
Yes! I'm thinking of taking a different approach for HTTPS anyways.